dynamic-unix.c

```diff
@@ -26,7 +26,8 @@ void * iio_dlsym(void *lib, const char *symbol)
 {
 	void *ptr;
 
+	dlerror();
 	ptr = dlsym(lib, symbol);
 
-	return ptr ? ptr : iio_ptr(-EINVAL);
+	return (ptr || !dlerror()) ? ptr : iio_ptr(-EINVAL);
```

Comment:
nit: maybe this would make it more simple?

`return dlerror() ? iio_ptr(-EINVAL) : ptr;`

Reply:
Well, is it in the spec that dlerror() returns NULL when dlsym() suceeds?

Reply:
`dlerror() returns NULL if no errors have occurred since initialization or since it was last called.`

I guess this is why we first call it before calling `dlysm()` so it's internal state is cleaned...

Reply:
Hmmmk. Fixed.